Content.Server/Chat/Systems/LastMessageBeforeDeathSystem.cs

```diff
@@ -0,0 +1,189 @@
+using System.Text;
+using Content.Server.Discord;
+using Content.Shared.Mobs.Systems;
+using System.Collections.Generic;
+using Content.Server.GameTicking;
+using Content.Server.Database;
+using Robust.Shared.Network;
+using Robust.Shared.Player;
+using Content.Server.Discord.Managers;
+using Robust.Shared.Configuration;
+using System.Threading.Tasks;
+using Content.Shared.CCVar;
+using Robust.Shared.Enums;
+
+namespace Content.Server.Chat.Systems
+{
+    internal class LastMessageBeforeDeathSystem : EntitySystem
+    {
+        [Dependency] private readonly IServerDbManager _db = default!;
+        [Dependency] private readonly IConfigurationManager _configManager = default!;
+        [Dependency] private readonly IEntityManager _entityManager = default!;
+        [Dependency] private readonly ILogManager _logManager = default!;
+        [Dependency] private readonly DiscordWebhook _discord = default!;
+
+        private MobStateSystem _mobStateSystem = default!;
+        private GameTicker _gameTicker = default!;
+        private IEntityManager _entManager = default!;
```

Comment:
You don't need this, all entity systems already have a reference with their `EntityManager` field.